examples/ios/cubic-demo/CubicAudioRecorder.swift

```diff
@@ -0,0 +1,206 @@
+            let newbuffer = AVAudioPCMBuffer(pcmFormat: downFormat!,
+                frameCapacity: AVAudioFrameCount(downFormat!.sampleRate * 0.4))
+                                            
+            let inputBlock: AVAudioConverterInputBlock = { (inNumPackets, outStatus) -> AVAudioBuffer? in
+                outStatus.pointee = AVAudioConverterInputStatus.haveData
+                let audioBuffer : AVAudioBuffer = buffer
+                return audioBuffer
+            }
+                                            
+            if let nb = newbuffer {
+                var error: NSError?
+                converter!.convert(to: nb, error: &error, withInputFrom: inputBlock)
+                let data = Data(buffer: nb, time: time)
+                self.cubicManager.sendStreamPartToServer(data, failure: nil)
+                _ = ExtAudioFileWrite(self.outref!, nb.frameLength, nb.audioBufferList)
+            }
+        })
+
+        try! audioEngine.start()
+        self.isRecording = true
+    }
+    
+    func stopStream() {
+        stopAudioEngine()
+        cubicManager.stopStream(failure: nil)
+    }
+    
+    private func stopAudioEngine() {
+        if audioEngine != nil && audioEngine.isRunning {
+            audioEngine.stop()
+            audioEngine.inputNode.removeTap(onBus: 0)
+            ExtAudioFileDispose(self.outref!)
+            try! AVAudioSession.sharedInstance().setActive(false)
+            self.isRecording = false
+        }
+    }
+    
+    @discardableResult
+    public func record() -> Bool {
+        guard let selectedModel = cubicManager.selectedModel else {
+            return false
+        }
+        
+        //kAudioFormatMPEG4AAC default
+        if !isAuthorized() {
+            return false
+        }
+        
+        let settings: [String : Any] = [
+            AVFormatIDKey: Int(kAudioFormatLinearPCM),
+            AVSampleRateKey: selectedModel.attributes.sampleRate,
+            AVNumberOfChannelsKey: 1,
+            AVLinearPCMBitDepthKey: 16,
+            AVLinearPCMIsFloatKey: false,
+            AVLinearPCMIsBigEndianKey: false,
+            AVEncoderAudioQualityKey: AVAudioQuality.high.rawValue
+        ]
+        
+        do {
+            let audioURL = CubicAudioRecorder.getWavURL()
+            audioRecorder = try AVAudioRecorder(url: audioURL, settings: settings)
+            audioRecorder.delegate = self
+            isRecording = audioRecorder.record()
+            return isRecording
+        } catch {
+            finishRecording(success: false)
+        }
+        
+        return false
+    }
+    
+    func finishRecording(success: Bool) {
+        audioRecorder?.stop()
+        audioRecorder = nil
+    }
+    
+    public func stop() {
+        if (isRecording) {
+            finishRecording(success: true)
+            let audioURL = CubicAudioRecorder.getWavURL()
+            cubicManager.uploadRecord(audioURL: audioURL, success: { (response) in
```

Comment:
Based on this, it seems that we are recording a file completely and then sending to cubic. Is that right? Or are we streaming in realtime as we record audio?